### PR TITLE
[csl] update CSL public APIs to use microseconds unit for period

### PR DESCRIPTION
**Pull request by abtink**
This commit updates public APIs for getting and setting CSL period to use microseconds unit instead of the internal ten symbols unit. This makes the APIs easier to use.

The CSL APIs have been renamed to follow the `otLinkGet/SetCsl{Item}()` pattern, which is the common naming style of OpenThread. This renaming will make the APIs more consistent and avoid potential confusion with the now-removed APIs, which used a different unit for CSL period.

This commit also updates the related CLI CSL commands:
- The `csl period` expects the given period to be in microseconds.
- The `csl` command (which outputs all CSL parameter) shows the CSL period in microsecond unit (e.g., "Period: 160000us").

The NCP spinel `SPINEL_PROP_THREAD_CSL_PERIOD` is also updated to use microsecond unit for CSL period.

The related test script are updated to use the new unit, in particular the test harness `setCSLperiod()` in `harness-thci/OpenThread.py` is updated (no need to convert from msec).

### CODE REVIEW

File: src/lib/spinel/spinel.h [hunk 1]
```diff
@@ -3055,10 +3055,13 @@ enum
     SPINEL_PROP_THREAD_NEW_DATASET = SPINEL_PROP_THREAD_EXT__BEGIN + 40,
 
     /// MAC CSL Period
-    /** Format: `S`
+    /** Format: `L`
```
Comment (edmont) [inline src/lib/spinel/spinel.h:3058]: Shouldn't this trigger a `SPINEL_RCP_API_VERSION` update?

**Review by edmont — COMMENTED** [review]
Thanks! A comment below.

File: src/lib/spinel/spinel.h (hunk 1 above)
Comment (abtink) [reply src/lib/spinel/spinel.h:3058]: Not really. We don't need to . This is only used in NCP (on MTD NCP) and it is not an RCP spinel property (not used by `RadioSpinel`). We need to increment `RCP_API_VERSION` when we change/add an spinel property that is used by RCP.

**Review by edmont — APPROVED** [review]

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**